Tests/SwiftCrossUITests/SwiftCrossUIInternalsTests.swift

```diff
@@ -0,0 +1,180 @@
+/// Required Imports
+import Foundation
+import Testing
+
+@testable import SwiftCrossUI
+
+@Suite(
+    "SwiftCrossUI's Internal Types",
+    .tags(.Internal)
+)
+struct InternalTests {
+
+    struct CounterView: View {
+        @State var count = 0
+
+        var body: some View {
+            VStack {
+                Button("Decrease") { count -= 1 }
+                Text("Count: 1")
+                Button("Increase") { count += 1 }
+            }.padding()
+        }
+    }
+
+    // NOTE: The Test ``CodableNavigationPath`` cannot be condensed into a parameterized test
+    // or a for loop without specifiying allowed Types individually. A Macro might be useful here
+    /// Makes sure that ``NavigationPath`` can retain data validating them with `compareComponents`.
+    @Test(
+        "The Codable NavigationPath can accurately store Codable data",
```

Comment:
Replace with `Ensures that NavigationPath instances can be round tripped to JSON and back`.